app/controllers/submissions_controller.rb

```diff
@@ -1,118 +1,81 @@
-  layout 'dashboard', only: [:valid, :rated, :to_rate, :rejected, :results]
-
-  def valid
-    submissions_valid = SubmissionRepository.new.valid
-
-    render :list, locals: { submission_presenters: create_submission_presenters(submissions_valid),
-      show_average: true, show_rates_count: true }
-  end
-
-  def rejected
-    submissions_rejected = SubmissionRepository.new.rejected
-
-    render :list, locals: { submission_presenters: create_submission_presenters(submissions_rejected),
-      show_average: false, show_rates_count: false }
-  end
-
-  def to_rate
-    submissions_to_rate = SubmissionRepository.new.to_rate
-
-    render :list, locals: { submission_presenters: create_submission_presenters(submissions_to_rate),
-      show_average: false, show_rates_count: true }
-  end
-
-  def results
-    submissions_accepted = SubmissionRepository.new.accepted
-    submissions_waitlist = SubmissionRepository.new.waitlist
-
-    render :results, locals: {
-      submissions_accepted: submissions_accepted,
-      submissions_waitlist: submissions_waitlist
-    }
-  end
 
   def show
     submission = Submission.find(params[:id])
-    submission_presenter = SubmissionPresenter.new(
-      submission,
-      submission.rates,
-      SubmissionRepository.new,
-      current_user
-    )
-
-    rate_presenters = create_rate_presenters(submission.rates)
-    comment_presenters = create_comment_presenters(submission.comments)
 
     render :show, locals: {
+      submission: SubmissionPresenter.build(submission, current_user),
+      answers: AnswerPresenter.collection(submission.answers),
       comment: Comment.new,
-      comment_presenters: comment_presenters,
-      rate_presenters: rate_presenters,
-      submission: submission_presenter
+      comment_presenters: CommentPresenter.collection(submission.comments),
+      rate_presenters: RatePresenter.collection(submission.rates)
     }
   end
 
   def new
     if Setting.preparation_period?
       render :preparation
     elsif Setting.registration_period?
-      submission = Submission.new
-      footer_presenter = FooterPresenter.new(Setting.get)
-      render :new, locals: { submission: submission, footer_presenter: footer_presenter }
+      render :new, locals: {
+        submission: Submission.new,
+        answers: build_form_answers(form_questions),
+        footer_presenter: FooterPresenter.new(Setting.get)
+      }
     else
       render :closed
     end
   end
 
   def thank_you
+    render :thank_you
   end
 
   def create
     submission = Submission.new(submission_params)
 
     if submission.valid?
-      submission_rejector = SubmissionRejector.new.reject_if_any_rules_broken(submission)
+      SubmissionRejector.new.reject_if_any_rules_broken(submission)
       submission.save
+      attributes_collection = answers_attributes(answers_params, submission.id)
+      Answer.create_collection(attributes_collection)
```

Comment:
Oh and if you follow this way, some unit test for this service would be useful ; ]